tests/pruners_tests/test_hyperband.py

```diff
@@ -130,3 +132,49 @@ def test_hyperband_max_resource_value_error():
 
     with pytest.raises(ValueError):
         _ = optuna.pruners.HyperbandPruner(max_resource="not_appropriate")
+
+
+def test_hyperband_filter_study():
```

Comment:
I think this is an optional request, but how about adding a test case to ensure that`_filter_study` does not apply any filters if the pruner is not `HyperbandPruner`? I think it will help developers to find future bugs in `_filter_study` although this pass is covered by many test cases.